tests/src/system/basic/WskBasicNodeTests.scala

```diff
@@ -80,6 +82,171 @@ class WskBasicNodeTests
+
+                    val duration = System.currentTimeMillis() - start
+                    duration should be >= expectedDuration.toMillis
+            }
+    }
+
+    it should "Ensure that whisk.trigger() still uses a callback when provided one" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            // this action supplies a 'next' callback to whisk.trigger()
+            val nameOfActionThatTriggers = "triggerAction"
+            val actionThatTriggers = Some(TestUtils.getTestActionFilename("triggerCallback.js"))
+
+            assetHelper.withCleaner(wsk.action, nameOfActionThatTriggers) {
+                (action, _) =>
+                    action.create(nameOfActionThatTriggers, actionThatTriggers)
+            }
+
+            // this is expected to fail this time because we have not yet created the trigger
+            val runReject = wsk.action.invoke(nameOfActionThatTriggers, Map("shouldSucceed" -> false.toJson))
+            withActivation(wsk.activation, runReject) {
+                activation =>
+                    activation.getFieldPath("response", "success") shouldBe Some(false.toJson)
+                    activation.getFieldPath("response", "result", "error") shouldBe defined
+            }
+
+            val triggerName = "UnitTestTrigger"
+
+            assetHelper.withCleaner(wsk.trigger, triggerName) {
+                (trigger, _) =>
+                    trigger.create(triggerName)
+            }
+
+            // now that we've created the trigger, running the action should succeed
+            val runResolve = wsk.action.invoke(nameOfActionThatTriggers, Map("shouldSucceed" -> true.toJson))
+            withActivation(wsk.activation, runResolve) {
+                activation =>
+                    activation.getFieldPath("response", "success") shouldBe Some(true.toJson)
+                    activation.getFieldPath("response", "result", "error") should not be defined
+            }
+    }
+
+    it should "Ensure that whisk.trigger() returns a promise" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            // this action supplies a 'next' callback to whisk.trigger()
+            val nameOfActionThatTriggers = "triggerAction"
+            val actionThatTriggers = Some(TestUtils.getTestActionFilename("triggerPromise.js"))
+
+            assetHelper.withCleaner(wsk.action, nameOfActionThatTriggers) {
+                (action, _) =>
+                    action.create(nameOfActionThatTriggers, actionThatTriggers)
+            }
+
+            // this is expected to fail this time because we have not yet created the trigger
+            val runReject = wsk.action.invoke(nameOfActionThatTriggers, Map("shouldSucceed" -> false.toJson))
+            withActivation(wsk.activation, runReject) {
+                activation =>
+                    activation.getFieldPath("response", "success") shouldBe Some(false.toJson)
```

Comment:
@csantanapr This test fails because `triggerPromise.js` is set up to resolve or reject based on the value of the `shouldSucceed` parameter. In this case, `shouldSucceed` is `false` (meaning I expect the trigger to fail because it doesn't exist yet), and the trigger fails (as expected) so `triggerPromise.js` _resolves_ with a message of "Trigger failed as expected." But here you can see the test is expecting the action to fail.

Obviously this is way too confusing 😉 

I will simplify the test actions to always reject when the trigger fails, and always resolve when the trigger succeeds. This will be much simpler and easy to understand, while still allowing me to test all paths.